File: Quivi.Application/Pos/Items/SessionItem.cs
namespace Quivi.Application.Pos.Items
{
    public record BaseSessionItem
    {
        public required int MenuItemId { get; init; }
        public decimal Quantity { get; init; }
        public decimal Price { get; init; }
    }

    public record SessionItem : BaseSessionItem
    {
        public decimal Discount { get; init; }
        public required IEnumerable<BaseSessionItem> Extras { get; init; }
    }

    public record PaidSessionItem : SessionItem
    {
        public decimal PaidQuantity { get; init; }
    }
}

File: quivi.pos.react.app/src/hooks/api/Dtos/sessions/SessionItem.ts
export interface BaseSessionItem {
    readonly menuItemId: string;
    readonly quantity: number;
    readonly price: number;
    readonly originalPrice: number;
}

export interface SessionItem extends BaseSessionItem {
    readonly id: string;
    readonly extras: BaseSessionItem[];
    readonly isPaid: boolean;
    readonly discountPercentage: number;
    readonly lastModified: string;
}